com.unity.render-pipelines.high-definition/Runtime/Material/AxF/ShaderPass/AxFSharePass.hlsl

```diff
@@ -17,7 +17,7 @@
 #if defined(_REQUIRE_UV2) || defined(_REQUIRE_UV3) || defined(DYNAMICLIGHTMAP_ON) || defined(DEBUG_DISPLAY) || (SHADERPASS == SHADERPASS_LIGHT_TRANSPORT)
 #define ATTRIBUTES_NEED_TEXCOORD2
 #endif
-#if defined(_REQUIRE_UV3) || defined(DEBUG_DISPLAY)
+#if defined(_REQUIRE_UV3) || defined(DEBUG_DISPLAY) || ((SHADERPASS == SHADERPASS_LIGHT_TRANSPORT) && defined(EDITOR_VISUALIZATION))
```

Comment:
I don't see any UV3 in the meta pass, are you sure you need this?  note that the meta pass used uv0, uv1 and uv2 but no uv3

Reply:
The meta pass needs VARYINGS_NEED_TEXCOORD3 defined. So I guess I should move this down below as follows:


#ifdef ATTRIBUTES_NEED_TEXCOORD3  || ((SHADERPASS == SHADERPASS_LIGHT_TRANSPORT) && defined(EDITOR_VISUALIZATION))
#define VARYINGS_NEED_TEXCOORD3
#endif

Reply:
Ah, VertMesh requires that uv3 exists if texCoord3 is used. So I will leave it as is if that's okay.

Reply:
can you point me where UV3 is used in meta pass, I haven't noticed it? thanks

Reply:
Line 65 of ShaderPassLightTransport.hlsl

https://github.com/Unity-Technologies/Graphics/pull/4533/files#diff-8a05acf5191b37d00df7c8de3aa78907221288291e6062afc6c3bd71b7905876

Reply:
Hmm... I think I'm misunderstanding something about how the v2f struct is constructed. I assumed that each texcoord is a float2, but now I'm getting errors in HDRP shadergraphs because the texcoords seem to be float4s. If the texcoords can be float4s, then I would only need up to texcoord2.

Reply:
ah, just assigning to .xy only fixes it. :) Can you check whether the usage of texcoords in ShaderPassLightTransport.hlsl is correct in latest?

Reply:
ah right, so indeed we will keep the support of uv3 because it can be used in shader graph (i.E an artists could create a graph with fetching albedo with uv3 ). So indeed my bad, I have forge't about this :)

Reply:
However you don't need to add the  ((SHADERPASS == SHADERPASS_LIGHT_TRANSPORT) as it is optional. it is IF the material use it then we will need to define it, so you should still remove this code

Reply:
Since the meta pass needs texcoord3, these lines in VertMesh.hlsl cause an error if uv3 does not also exist:

#if defined(VARYINGS_NEED_TEXCOORD3) || defined(VARYINGS_DS_NEED_TEXCOORD3)
    output.texCoord3 = input.uv3;
#endif